src/main/java/frc/robot/sensors/Pigeon.java

```diff
@@ -0,0 +1,27 @@
+package frc.robot.sensors;
+import com.ctre.phoenix.sensors.*;
+import frc.robot.subsystems.DriveBase;
+
+public class Pigeon {
+
+    static PigeonIMU _pidgey;      // Pigeon IMU used to enforce straight drive
+
+    static {
+        //get talon
+        _pidgey = new PigeonIMU(DriveBase.getInstance().leftFrontTalon);
+
+        //setup gyro angle finder
+        _pidgey.setFusedHeading(0.0, 30);
+    }
+
+    public static PigeonIMU getPigeon() {
+        return _pidgey;
+    }
+
+    public static double getCurrentAngle(){
+        //Gets the current angle
+        PigeonIMU.FusionStatus fusionStatus = new PigeonIMU.FusionStatus();
```

Comment:
Should this be `_pidgey` instead of creating a new object?

Reply:
Agreed. Let's be very careful about instantiating multiple objects that tie back to a single piece of physical hardware. At worst, there will be a robot code crash (this used to happen to us a lot when we were allocating IOs on the RoboRio). At best this will lead to subtle bugs because one instance will tell the hardware to do X or keep track of Y and another instance will have no such knowledge and your code will get nulls or bad data or worse.

Reply:
No this code is fine. The Pigeon library is very C-style where you instantiate a class and pass it into a function which modifies the class instead of returning a result

Reply:
OK, as long as that is clear to everyone. 

This is a general comment but remember that we are writing this code not just for this year but potentially for future years as well, including for rookie students. For that reason I would prioritize things like consistence, readability, clarity, and simplicity over clever tricks and special exceptions.